### PR TITLE
Extra gatherers and utils

### PR DESCRIPTION
**Pull request by timclifford**

### CODE REVIEW

File: gatherers/EnvVarGatherer.go [hunk 1]
```diff
@@ -20,10 +20,11 @@ func (p *envVarGatherer) GatherFacts() ([]GatheredFact, error) {
 	}
 	return []GatheredFact{
 		{
-			Name: "LAGOON_VERSION",
+			Name: "lagoon",
 			Value: lagoonVersion,
 			Source: "env",
-			Description: "This is the current LAGOON_VERSION running in the environment",
+			Description: "This is the current version of Lagoon",
```
Comment (tobybellwood) [inline gatherers/EnvVarGatherer.go:26]: Here the Lagoon version relates specifically to the release of the base image the fact is run inside yes?

File: utils/yaml-parser.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package utils
+
+import (
+	"errors"
+	"gopkg.in/yaml.v2"
+)
+
+type DockerComposeConfig struct {
```
Comment (tobybellwood) [inline utils/yaml-parser.go:8]: These x-* values don't always exist in docker-compose files - what are we using them for?

File: gatherers/DockerComposeGatherer.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+	config, err := utils.UnmarshallYamlToStructure(lagoonConfigBytestream)
+	if err != nil {
+		log.Fatalf("There was an issue unmarshalling the docker-compose.yml file: %s", err)
+	}
+
+	if config.Services != nil {
+		applies = true
+
+		for k, element := range config.Services {
```
Comment (tobybellwood) [inline gatherers/DockerComposeGatherer.go:36]: important to note that any service with a lagoon.type=none will not be in Lagoon, but will still show up here - maybe we need to think about whether this is the right place to collect services from.

File: gatherers/DockerComposeGatherer.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+			serviceItems := element.(map[interface {}]interface {})
+
+			var service dockerComposeService
+			service.Name = k
+			service.Description = "Service found in docker-compose.yml"
+
+			for j, val := range serviceItems {
+				if str, ok := val.(string); ok {
+					if j == "image" {
+						service.Image = str
```
Comment (tobybellwood) [inline gatherers/DockerComposeGatherer.go:46]: Additionally, collecting the service.Image will be overridden for DBaaS-type services, which won't be apparent here?

File: utils/yaml-parser.go (hunk 2 above)
Comment (timclifford) [reply utils/yaml-parser.go]: Yeah on reflection, grabbing these kind of values from here isn't really useful (as its more for local dev config)

File: gatherers/DockerComposeGatherer.go (hunk 3 above)
Comment (timclifford) [reply gatherers/DockerComposeGatherer.go:42]: I'm thinking the docker-compose gatherer itself might not be needed

**Pull request merged**